From b222ae11c3435bd30997a30b9ede9659392735b4 Mon Sep 17 00:00:00 2001
From: Mikolaj <mikolaj.konarski@gmail.com>
Date: Wed, 15 May 2013 15:41:32 +0200
Subject: [PATCH 7/9] FIx to "Add last commit short hash"

The ghc toolchain doesn't cope with ghc versions other than decimal
numbers separated by dots, so instead of a commit hash, we append
the number of commits since the last tag matching "*release",
e.g., we get a ghc version '7.4.2.6'.
---
 aclocal.m4 | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index 4de9954..c1ef357 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1461,13 +1461,15 @@ AC_SUBST([GccExtraViaCOpts],$fp_cv_gcc_extra_opts)
 AC_DEFUN([FP_SETUP_PROJECT_VERSION],
 [
 if test "$RELEASE" = "YES"; then
-    AC_MSG_CHECKING([for GHC version date])
+    AC_MSG_CHECKING([for GHC version patchlevel])
     if test -f VERSION_DATE; then
         PACKAGE_VERSION=${PACKAGE_VERSION}.`cat VERSION_DATE`
         AC_MSG_RESULT(given $PACKAGE_VERSION)
     elif test -d .git; then
-        ver_date=`git rev-parse --short HEAD`
-        PACKAGE_VERSION=${PACKAGE_VERSION}.$ver_date
+        changequote(, )dnl
+        ver_patchlevel=`git describe --tags --match "*release" | sed 's/^.*release-\([0-9]*\)-.*$/\1'/`
+        changequote([, ])dnl
+        PACKAGE_VERSION=${PACKAGE_VERSION}.$ver_patchlevel
         AC_MSG_RESULT(inferred $PACKAGE_VERSION)
     elif test -d _darcs; then
         # TODO: Remove this branch after conversion to Git
-- 
1.7.12.4 (Apple Git-37)

